src/binder/bind/bind_reading_clause.cpp

```diff
@@ -137,86 +137,136 @@ std::unique_ptr<BoundReadingClause> Binder::bindLoadFrom(
     if (readerConfig->getNumFiles() > 1) {
         throw BinderException("Load from multiple files is not supported.");
     }
-    std::vector<std::string> inputColumnNames;
-    std::vector<std::unique_ptr<common::LogicalType>> inputColumnTypes;
+    // Bind columns from input.
+    std::vector<std::string> expectedColumnNames;
+    std::vector<std::unique_ptr<common::LogicalType>> expectedColumnTypes;
     for (auto& [name, type] : loadFrom.getColumnNameDataTypesRef()) {
-        inputColumnNames.push_back(name);
-        inputColumnTypes.push_back(bindDataType(type));
+        expectedColumnNames.push_back(name);
+        expectedColumnTypes.push_back(bindDataType(type));
     }
+    // Detect columns from file.
     std::vector<std::string> detectedColumnNames;
     std::vector<std::unique_ptr<common::LogicalType>> detectedColumnTypes;
-    switch (fileType) {
+    sniffFiles(*readerConfig, detectedColumnNames, detectedColumnTypes);
+    // Validate and resolve columns to use.
+    expression_vector columns;
+    if (expectedColumnTypes.empty()) { // Input is empty. Use detected columns.
+        columns = createColumnExpressions(*readerConfig, detectedColumnNames, detectedColumnTypes);
+    } else {
+        validateNumColumns(expectedColumnTypes.size(), detectedColumnTypes.size());
+        if (fileType == common::FileType::PARQUET) {
+            validateColumnTypes(expectedColumnNames, expectedColumnTypes, detectedColumnTypes);
+        }
+        columns = createColumnExpressions(*readerConfig, expectedColumnNames, expectedColumnTypes);
+    }
+    auto info = std::make_unique<BoundFileScanInfo>(
+        std::move(readerConfig), std::move(columns), nullptr, TableType::UNKNOWN);
```

Comment:
Can we get a comment on what `nullptr` is here?

Reply:
Done